Add Aqua.jl for some static testing

lgoettgens opened the pull request:
Resolves #1964.

Three of the performed tests (`ambiguities`, `unbound_args`, and `piracy`) currently fail and are thus disabled (with a comment to check if/with how much work they can be fixed).

`deps_compat` requires all dependencies to have a `[compat]` entry, so I added the missing ones. The JuliaRegistry currently does not require those for JLLs (cf. the last paragraph of https://juliaregistries.github.io/RegistryCI.jl/stable/guidelines/#Upper-bounded-[compat]-entries), but adding them should not hurt.

`stale_deps` fails on my local setup with some weird error containing the UUID of `GAP_lib_jll`. This can be investigated in a separate PR.

lgoettgens committed:
Add Aqua.jl for some static testing

fingolfin commented on Project.toml:
```diff
@@ -26,6 +26,7 @@ cohomCalg_jll = "5558cf25-a90e-53b0-b813-cadaa3ae7ade"
 [compat]
 AbstractAlgebra = "0.29.3"
 AlgebraicSolving = "0.3.0"
+Aqua = "0.6"
```
Do we really need this as a dependency when it is only used by the tests?

lgoettgens replied:
It is not a dependency but only listed in the `[extra]`. With the compact entry, we can make sure that an `Aqua.jl` release (that probably will change some of the performed tests) does not break our CI.

lgoettgens replied:
At least in theory. I could unfortunately not find a statement in the docs if `[compact]` is used at all for `[extra]` dependencies.

fingolfin replied:
yeah, I misphrased that, sorry

Anyway, I think we should move away from the old-style usage of `[extras]` etc in the main `Project.toml` and instead move to a dedicated `test/Project.toml`, that would neatly resolve this issue.

lgoettgens replied:
I strongly agree. But I think this should be done in a separate PR. I created #2374 to remember.

thofma replied:
What exactly is the issue? I don't see a problem that is solved by this.

lgoettgens replied:
> What exactly is the issue? I don't see a problem that is solved by this.

`Aqua.jl` may release a new version `v0.7` in the future that includes some additional test, which fails for OSCAR.
Without this, the second `v0.7` is released, our CI breaks (on all branches). With this change, we can change the used `Aqua.jl` version ourselves, in a controlled environment/PR.

Note:
> At least in theory. I could unfortunately not find a statement in the docs if [compact] is used at all for [extra] dependencies.

thofma replied:
Yes, the `[compat]` entry is also valid for the `[extra]` entries, which you did here. So everything is fine.

fingolfin replied:
@lgoettgens agreed (about the separate PR)

lgoettgens committed:
Move `Aqua` tests to its own file

to enable use of `Oscar.test_module` and the printing of used time per test in the CI.

fingolfin reviewed (commented):
Fine by me
